### PR TITLE
Added authentication pages and api calls

### PR DESCRIPTION
**Pull request by yaswanthmyneni**
# Added authentication pages and API calls

## Preview: [lucent-video-lib](https://deploy-preview-7--ym-video-lib-dev.netlify.app/)
-----> Click on the categories on the homepage to visit the video listing page in the preview.
-----> On clicking the watch now button, it will be redirected to the video page and also adds the data to the history page at the same time.
-----> login before adding any data to watch later, playlist, history, like pages.

### CODE REVIEW

**Comment by anasm11** [comment]
The ui and code structure look clean. Good to merge.

**Review by anasm11 — APPROVED** [review]
The ui and code structure look clean. Good to merge.

File: src/pages/authentication-pages/signin-page/SignInPage.jsx [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+            <label htmlFor="email">Email Id</label>
+            <input
+              className="input"
+              type="email"
+              id="email"
+              name="email"
+              required
+              onChange={(e) =>
+                authDispatch({ type: "EMAIL", payload: e.target.value })
+              }
```
Comment (Naman-Saxena1) [inline src/pages/authentication-pages/signin-page/SignInPage.jsx:36]: I am not sure if this is fine or bad practice, you are using onChange without a value attribute on the input field. I will read more about it.

**Review by Naman-Saxena1 — CHANGES_REQUESTED** [review]
Code looks fine 👍 
For UI I think on sidebar you can add home page options, it will make it more user friendly.

**Pull request merged**